scrapyd_k8s/api.py

```diff
@@ -148,5 +189,11 @@ def run():
     if config_username is not None and config_password is not None:
         enable_authentication(app, config_username, config_password)
 
+    pod_watcher_thread = threading.Thread(
+        target=watch_pods,
+        kwargs={'namespace': scrapyd_config.get('namespace', 'default')}
+    )
+    pod_watcher_thread.start()
+
```

Comment:
It's maybe a bit early to start about performance-related things, but I think that the job logging feature will not be used by everyone, and adds overhead: for every change in pod state for every pod in the namespace, this Python program examines it to decide whether to do anything with it. So let's only start the pod watcher if we will actually use it (so don't when the job logging is not enabled, or if there is a configuration issue with it that allows scrapyd-k8s to run but the job logging to not be active - if that situation could happen).

Reply:
Combined with the previous comment of moving the functionality that does something with the watched pod info, the code here could ask the job logging code if it needs the pod watcher or not (depending on its configuration).